draft-decraeneginsberg-lsr-isis-fast-flooding.xml

```diff
@@ -345,11 +345,11 @@
         LSPs be acknowledged more promptly as well. This requires a reduction
         in the delay in sending PSNPs.</t>
 
-		<t>Receiver SHOULD reduce partialSNPInterval. The choice of this lower value is a local choice. It may depend on the (available) processing power of the node, the number of adjacencies, the requirement to synchronize the LSDB more quickly. 200 ms seems a reasonable value.</t>
-		<t>In addition to the timer based partialSNPInterval, the receiver SHOULD keep track of the number of unacknowledged LSPs per circuit and level. When this number exceeds a preset threshold LSP per PSNP (LPP), the receiver SHOULD immediately send a PSNP without waiting for the PSNP timer to expire. In case of a burst of LSPs, this allows for more frequent PSNPs, hence a faster feedback loop to the sender. In the absence of burst, the usual time-based PSNP approach comes into effect. If InterfaceLSPReceiveWindow is 	advertised, LPP SHOULD be lower and best performance is achieved when LPP is an integer fraction of InterfaceLSPReceiveWindow. This number SHOULD also be lower or equal to 90 as this is the maximum number of LSPs that can be acknowledged in a PSNP, hence waiting longer would not reduce the number of PSNPs sent but would delay the acknowledgements. Based on experimental evidence, 15 unacknowledged LSPs is a right value assuming that InterfaceLSPReceiveWindow is at least twice bigger (>30).</t>
+		<t>The receiver SHOULD reduce its partialSNPInterval. The choice of this lower value is a local choice. It may depend on the available processing power of the node, the number of adjacencies, and the requirement to synchronize the LSDB more quickly. 200 ms seems to be a reasonable value.</t>
+		<t>In addition to the timer based partialSNPInterval, the receiver SHOULD keep track of the number of unacknowledged LSPs per circuit and level. When this number exceeds a preset threshold of LSPs Per PSNP (LPP), the receiver SHOULD immediately send a PSNP without waiting for the PSNP timer to expire. In case of a burst of LSPs, this allows for more frequent PSNPs, giving faster feedback to the sender. Outside of the burst case, the usual time-based PSNP approach comes into effect. If an LSP Burst Window is advertised, LPP SHOULD be lower and the best performance is achieved when LPP is an integer fraction of the LSP Burst Window. This number SHOULD also be lower or equal to 90 as this is the maximum number of LSPs that can be acknowledged in a PSNP, hence waiting longer would not reduce the number of PSNPs sent but would delay the acknowledgements. Based on experimental evidence, 15 unacknowledged LSPs is a good value assuming that the LSP Burst Window is at least 30.</t>
 		<t>By deploying both the time-based and the threshold-based PSNP approaches, the receiver can be adaptive to both LSP bursts and infrequent LSP updates.  </t>
 
```

Comment:
Replace the prior two paragraphs with:

"The ability to safely do faster flooding benefits from faster
acknowledgments. It is RECOMMENDED that partialSNPInterval be reduced
to a value less than 500 milliseconds. More aggressive values MAY
result in better throughput and SHOULD be considered. The value chosen
may depend on the available processing power of the node, the number of
adjacencies, and the desired flooding speed."

Why do I suggest this? Because in our testing we have been able to
increase flooding speed even when the receiver uses a slow PSNP interval.
So the use of "necessary" seems too strong. The "necessity" depends in part
on the approach chosen to implement faster flooding.

I'd also prefer that the discussion of the relationship between LPP
and Burst Window be moved to Section 6.2 as it applies to that Example
algorithm. Keep the discussion of "90" as an upper bound as that is
based on MTU - not algorithm choice.

Reply:
I think that the LPP improvement should be kept. Arista has shown that this improve the flooding performance even when RWIN is not implemented on the sender side.
One editorial option could be to tie that fast ack to the advertisement of a RWIN: a router advertising a RWIN value SHOULD fast ack using a delay below 500ms and every LPP LSP.
Also, a priori all algo benefit from faster feedback hence faster ack (unless one hard code special value/delay such as re-adapting the sending rate every second)

Reply:
So actually you are only asking to move the sentence "If InterfaceLSPReceiveWindow is advertised, LPP
   SHOULD be lower and best performance is achieved when LPP is an
   integer fraction of InterfaceLSPReceiveWindow."? I though you were asking to remove the whole LPP paragraph.
I can live with moving that sentence to the algo specific section, but that would mean adding a normative sentence in that section.
Note that if your implementation does not use Burst at all, you are not impacted by the sentence.

Reply:
I'm afraid that I don't understand all of the objections here.

Ok, I've removed the word 'necessary'.

The discussion of LPP and bursts is orthogonal to the choice of algorithm.  Bursts happen in every sane implementation and implementing LPP is a huge win even if you make no further changes.

I've done a small amount of rewording and will leave it at that for now. You are welcome, of course, to make further changes in your pass.

Reply:
Perhaps this is best addressed as an open issue - not blocking for first draft...but my point is this:

"If an LSP Burst Window is advertised,
   LPP SHOULD be lower and the best performance is achieved when LPP is
   an integer fraction of the LSP Burst Window."

The statement that "best performance is achieved..." applies only to Algorithm #1. Since Algorithm #2 doesn't use burst in its logic, the idea that this behavior will result in better/best performance does not apply.  So I would like this text to go into the section for the algorithm to which it applies.

That is the main point I am trying to make.
But we can stop trying to resolve this now and come back to it later...
Thanx.

Reply:
We have heard your point.  

With your algorithm, assuming that the receiver doesn't advertise a burst window, the statement is vacuously true.

Whether or not a burst window is advertised, it is still optimal to keep the LPP an integer fraction of the burst window so that the transmitter can continue to burst while it is processing acknowledgments. In fact, in an optimal scenario, the ratio of LPP to burst window will allow the transmitter to stay in a sustained burst mode. The exact value of that ratio is still TBD, but I suspect that it's something like 4. :-)

I already resolved this conversation so that we could move forward and asked that we not continue unless there were unacceptable objections. You continue to argue.

If you are willing to stop trying to resolve this now, then you need to stop arguing.